backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ParentNodeFinder.java

```diff
@@ -0,0 +1,83 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.syson.diagram.common.view.services;
+
+import java.util.stream.Stream;
+
+import org.eclipse.sirius.components.diagrams.Diagram;
+import org.eclipse.sirius.components.diagrams.Node;
+
+/**
+ * Used to find parent nodes inside a given Diagram.
+ *
+ * @author Jerome Gout
+ */
+public class ParentNodeFinder {
+
+    private final Diagram diagram;
+
+    public ParentNodeFinder(Diagram diagram) {
+        super();
+        this.diagram = diagram;
```

Comment:
Could you add an Objects.requireNonNull here please?

Reply:
Done